### PR TITLE
LPS-200954 Add support for using Datetime and Boolean in the Object Proxy Salesforce filter

### PR DESCRIPTION
**Pull request by jairmedeiros**

### CODE REVIEW

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java [hunk 1]
```diff
@@ -127,7 +128,12 @@ public Object visitLiteralExpression(LiteralExpression literalExpression)
 		if (!Objects.equals(
 				LiteralExpression.Type.DATE, literalExpression.getType()) &&
 			!Objects.equals(
-				LiteralExpression.Type.STRING, literalExpression.getType())) {
+				LiteralExpression.Type.STRING, literalExpression.getType()) &&
```
Comment (gabrielwas) [inline modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java:131]: @jairmedeiros could you sort it? The following order: BOOLEAN, DATE, DATE_TIME, STRING

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java [hunk 2]
```diff
@@ -140,6 +146,12 @@ public Object visitLiteralExpression(LiteralExpression literalExpression)
 				StringPool.APOSTROPHE);
 		}
 
+		if (Objects.equals(
```
Comment (gabrielwas) [inline modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java:149]: Can you sort it? Put this if checking before the string one. Thx

**Review by gabrielwas — CHANGES_REQUESTED** [review]
@jairmedeiros Could you check the comments, pls? Thx

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java (hunk 1 above)
Comment (jairmedeiros) [reply modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java]: done!

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java (hunk 2 above)
Comment (jairmedeiros) [reply modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java:147]: done!

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
ci:forward:force

**Comment by guilhermedcamacho** [comment]
ci:forward:force

**Pull request closed without merging**